activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb

```diff
@@ -48,9 +48,7 @@ def visit_TableDefinition(o)
               statements.concat(o.indexes.map { |column_name, options| index_in_create(o.name, column_name, options) })
             end
 
-            if supports_foreign_keys_in_create?
```

Comment:
This isn't true for the abstract adapter, so I think we still need the conditional for now. (Or, if we're okay to declare that a baseline requirement, it'd need to change there too.)